### ISSUE TITLE
I so want to use this for retrying `cy.task`

### ISSUE THREAD

**Issue by bahmutov** [NicholasBoll/cypress-pipe#7]
but it only evaluates functions that return cypress chainers once. 

I have done my own little code experiment in https://github.com/bahmutov/retry-task-demo (which found a problem right away https://github.com/cypress-io/cypress/issues/3144) but it is working.

**Comment by NicholasBoll** [comment]
@bahmutov, do you think `cypress-pipe` should retry all funtions that contain Cypress chainers? I thought about it and decided not to for 2 reasons off the top of my head - retrying side effect things like `click` which have their own timeouts and the command log getting spammed with cy logs.

There are probably other edge cases as well. I could always try it to see what happens. You could probably experiment by removing the retry restriction.

**Comment by bahmutov** [comment]
I definitely see value in retrying everything even things that have side-effects - especially if the user really wants to. Real world problem solved here: https://github.com/cypress-io/cypress/issues/3135#issuecomment-454445249

**Comment by NicholasBoll** [comment]
Retrying is handled by `verifyUpcomingAssertions`. I dug around there a bit to see why assertions don't retry from functions that contain Cypress commands. It is quite complex and hard to follow. Do you have any insights. It looks like that code was last touched by Brian over a year ago.
